### PR TITLE
Sync shadow presets support in theme.json

### PR DESCRIPTION
**Pull request by Mamaduka**
Trac ticket: https://core.trac.wordpress.org/ticket/57559

Gutenberg PRs:

- https://github.com/WordPress/gutenberg/pull/46813
- https://github.com/WordPress/gutenberg/pull/47272

### CODE REVIEW

**Comment by Mamaduka** [comment]
I just pushed the `theme.json` update for shadow presets. Based on the diff, this is the only change since WP 6.1, so I won't create a separate PR.

Cc @oandregal

**Comment by oandregal** [comment]
> I just pushed the theme.json update for shadow presets. Based on the diff, this is the only change since WP 6.1, so I won't create a separate PR.

:+1: Good call, those changes pertain to this feature as well.

**Review by oandregal — APPROVED** [review]
Code changes look good, it has tests, and I verified that it works as expected following the test instructions at https://github.com/WordPress/gutenberg/pull/46813

**Comment by hellofromtonya** [comment]
Current reviewing. There are a few tweaks needed to the tests. And there's a merge conflict.

**Comment by hellofromtonya** [comment]
Rebase this PR on top of `wordpress-develop` current `trunk` branch to resolve the merge conflict. Force pushed.

File: tests/phpunit/tests/theme/wpThemeJson.php [hunk 1]
```diff
@@ -4434,4 +4434,75 @@ public function test_get_stylesheet_returns_outline_styles() {
...
+							),
+						),
+					),
+				),
+			)
+		);
+
+		$styles = 'body{--wp--preset--shadow--natural: 5px 5px 5px 0 black;--wp--preset--shadow--sharp: 5px 5px black;}';
+		$this->assertEquals( $styles, $theme_json->get_stylesheet(), 'Returned of "::get_stylesheet" does not match expectations' );
+		$this->assertEquals( $styles, $theme_json->get_stylesheet( array( 'variables' ) ), 'Returned of "::get_stylesheet" does not match expectations' );
```
Comment (hellofromtonya) [inline tests/phpunit/tests/theme/wpThemeJson.php]: * use `assertSame()`
* Failure messages need to be different to identify which assertion failed.

Commit coming.

File: tests/phpunit/tests/theme/wpThemeJson.php [hunk 2]
```diff
@@ -4434,4 +4434,75 @@ public function test_get_stylesheet_returns_outline_styles() {
...
+					),
+				),
+			)
+		);
+
+		$global_styles  = 'body{--wp--preset--shadow--natural: 5px 5px 0 0 black;}body { margin: 0; }.wp-site-blocks > .alignleft { float: left; margin-right: 2em; }.wp-site-blocks > .alignright { float: right; margin-left: 2em; }.wp-site-blocks > .aligncenter { justify-content: center; margin-left: auto; margin-right: auto; }';
+		$element_styles = 'a:where(:not(.wp-element-button)){box-shadow: var(--wp--preset--shadow--natural);}.wp-element-button, .wp-block-button__link{box-shadow: var(--wp--preset--shadow--natural);}p{box-shadow: var(--wp--preset--shadow--natural);}';
+		$styles         = $global_styles . $element_styles;
+
+		$this->assertEquals( $styles, $theme_json->get_stylesheet() );
```
Comment (hellofromtonya) [inline tests/phpunit/tests/theme/wpThemeJson.php]: * use `assertSame()`

Commit coming.

**Review by hellofromtonya — APPROVED** [review]
* Confirmed both GB PRs are present in this PR ✅ 
* Changes are noted in `@since 6.2.0`  ✅ 
* Includes tests  ✅ 
* Has a test report  ✅ 

Ready for commit.

File: tests/phpunit/tests/theme/wpThemeJson.php (hunk 2 above)
Comment (hellofromtonya) [reply tests/phpunit/tests/theme/wpThemeJson.php]: Resolved in https://github.com/WordPress/wordpress-develop/pull/3915/commits/5e3e53642a639a3a6d07e8d0f9dbc086b96c68c9.

File: tests/phpunit/tests/theme/wpThemeJson.php (hunk 1 above)
Comment (hellofromtonya) [reply tests/phpunit/tests/theme/wpThemeJson.php]: Resolved in https://github.com/WordPress/wordpress-develop/pull/3915/commits/5e3e53642a639a3a6d07e8d0f9dbc086b96c68c9.

**Comment by hellofromtonya** [comment]
Committed via https://core.trac.wordpress.org/changeset/55176.

**Pull request closed without merging**